### PR TITLE
[dvsim] Improve --help message

### PR DESCRIPTION
**Pull request by rswarbrick**
Improve --help message for dvsim.py. This is a series of 3 patches. The first two are minor cleanups; the third is the large patch, but should have no functional change - it's just shuffling arguments around and adding help strings.

### CODE REVIEW

File: util/dvsim/dvsim.py [hunk 1]
```diff
@@ -109,319 +118,307 @@ def sigint_handler(signal_received, frame):
...
-    parser.add_argument("--xprop-off",
-                        default=False,
+    parser.add_argument("--max-parallel", "-mp",
+                        type=int,
+                        default=16,
+                        metavar="N",
+                        help=('Run only up to N builds/tests at a time. '
+                              'Default value 16.'))
+
+    whatg = parser.add_argument_group('Choosing what to run')
```
Comment (sriyerg) [inline util/dvsim/dvsim.py:212]: This group should be moved to the top. This is pretty much what a new user would look for.

File: util/dvsim/dvsim.py [hunk 2]
```diff
@@ -109,319 +118,307 @@ def sigint_handler(signal_received, frame):
...
+                        const="default",
+                        metavar="D",
+                        help=('With no argument, print verbose dvsim tool '
+                              'messages. With --verbose debug, the volume of '
+                              'messages is even higher.'))
+
+    parser.add_argument("--dry-run", "-n",
+                        action='store_true',
+                        help=("Print dvsim tool messages but don't actually "
+                              "run any command"))
```
Comment (sriyerg) [inline util/dvsim/dvsim.py:182]: This should be in its own group "dvsim options" or sth and moved to the end.

File: util/dvsim/dvsim.py [hunk 3]
```diff
@@ -109,319 +118,307 @@ def sigint_handler(signal_received, frame):
...
+                              "be shown at the end."))
 
-    parser.add_argument("-d",
-                        "--dump",
-                        default="fsdb",
-                        metavar="fsdb|shm",
-                        help="Dump waves in fsdb or shm.")
+    parser.add_argument("--publish",
+                        action='store_true',
+                        help="Publish results to reports.opentitan.org.")
```
Comment (sriyerg) [inline util/dvsim/dvsim.py:197]: Group this into "Generating and publishing results"

File: util/dvsim/dvsim.py [hunk 4]
```diff
@@ -109,319 +118,307 @@ def sigint_handler(signal_received, frame):
...
-
-    parser.add_argument("-p",
-                        "--profile",
-                        default="none",
-                        metavar="time|mem",
-                        help="Turn on simulation profiling")
-
-    parser.add_argument("--xprop-off",
-                        default=False,
+    parser.add_argument("--max-parallel", "-mp",
```
Comment (sriyerg) [inline util/dvsim/dvsim.py:205]: Group this and `--job-prefix` into "Job dispatch options"

**Comment by rswarbrick** [comment]
Hi @sriyerg. I've split non-documentation changes into a separate PR as requested. As you'd expect, the two patches clash all over the place, which we can't deal with nicely by rebasing because the commits are in different PRs.

Let's park this PR until that's merged, then I'll rebase this one on the resulting change and address your other comments here.

File: util/dvsim/dvsim.py [hunk 5]
```diff
@@ -2,14 +2,19 @@
...
 # SPDX-License-Identifier: Apache-2.0
-"""
-dvsim is a command line tool to deploy regressions for design verification. It uses hjson as the
-format for specifying what to build and run. It is an end-to-end regression manager that can deploy
-multiple builds (where some tests might need different set of compile time options requiring a
-uniquely build sim executable) in parallel followed by tests in parallel using the load balancer
-of your choice. dvsim is built to be tool-agnostic so that you can easily switch between tools
-available at your disposal. dvsim uses fusesoc as the starting step to resolve all inter-package
-dependencies and provide us with a filelist that will be consumed by the sim tool.
+"""dvsim is a command line tool to deploy regressions for design verification.
```
Comment (sriyerg) [inline util/dvsim/dvsim.py:5]:
```suggestion
"""dvsim is a command line tool to deploy ASIC tool flows such as regressions for design verification (DV), formal property verification (FPV), linting and synthesis.
```

**Comment by rswarbrick** [comment]
Force-push rebases now that the argument handling PR has been merged. No other change.

**Pull request merged**